src/transport/advection_diffusion_model.hh

```diff
@@ -147,9 +147,9 @@ public:
 	 */
 	virtual void compute_source_coefficients(const std::vector<arma::vec3> &point_list,
 			const ElementAccessor<3> &ele_acc,
-			std::vector<arma::vec> &sources_conc,
-			std::vector<arma::vec> &sources_density,
-			std::vector<arma::vec> &sources_sigma) = 0;
+			std::vector<std::vector<double> > &sources_conc,
+			std::vector<std::vector<double> > &sources_density,
+			std::vector<std::vector<double> > &sources_sigma) = 0;
```

Comment:
I think it makes sense to have the substance loop inside the loop over elements, since in each element we call some "reinit" method which would have to be otherwise called for each substance repeatedly. Of course there is no problem to change compute_* methods to work with single substance only, although the current version may be slightly more efficient (see the methods in ConcentrationTransportModel).